pkg/proxy/iptables/proxier.go

```diff
@@ -1126,24 +1126,22 @@ func (proxier *Proxier) syncProxyRules() {
 				)
 
 				destChain := svcXlbChain
-				// We have to SNAT packets to external IPs if externalTrafficPolicy is cluster.
+				// We have to SNAT packets to external IPs if externalTrafficPolicy is cluster
+				// and the traffic is NOT Local. Local traffic coming from Pods and Nodes will
+				// be always sent directly to the corresponding Service, so no need to SNAT
 				if !svcInfo.OnlyNodeLocalEndpoints() {
 					destChain = svcChain
-					writeLine(proxier.natRules, append(args, "-j", string(KubeMarkMasqChain))...)
+					if proxier.localDetector.IsImplemented() {
+						// This masquerades off-cluster traffic to a External IP. The idea
+						// is that you can establish a static route for your Service range,
+						// routing to any node, and that node will bridge into the Service
+						// for you.  Since that might bounce off-node, we masquerade here.
```

Comment:
You start out talking about External IPs but then switch to talking about regular service IPs...

Reply:
I copied pasted from the other comment, I have to change it , it doesn't read well ....

Reply:
yeah, the comment threw me too